substrate/state-machine/src/ext.rs

```diff
@@ -126,8 +126,9 @@ where
 	H::Out: Ord + Encodable
 {
 	fn storage(&self, key: &[u8]) -> Option<Vec<u8>> {
-		self.overlay.storage(key).map(|x| x.map(|x| x.to_vec())).unwrap_or_else(||
-			self.backend.storage(key).expect("Externalities not allowed to fail within runtime"))
+		use {try_read_overlay_value};
+		try_read_overlay_value(self.overlay, self.backend, key)
+			.expect("Externalities not allowed to fail within runtime")
```

Comment:
we can probably just do `.map_err(|e| error!(target: "state-machine", "Externalities call failed! {:?}", e)).ok().and_then(|inner| inner)`, which then will at least avoid the panic. better would be to shuffle the `Err` through into the wasm interpreter so the execution can be early-exited. not sure if that's possible yet, though.